packages/flutter/test/widgets/media_query_test.dart

```diff
@@ -564,4 +564,146 @@ void main() {
     expect(outsideBoldTextOverride, false);
     expect(insideBoldTextOverride, true);
   });
+
+  group('MediaQueryData parameters cannot be null', () {
+
+    String generateErrorMessage(String parameterName) {
+      return '\'$parameterName != null\': is not true';
+    }
+
+    test('size parameter', () {
+      try {
+        MediaQueryData(size: null);
+        fail('The assert was never called when it should have been');
+      } on AssertionError catch (error) {
+        expect(error.toString(), contains(generateErrorMessage('size')));
+      }
+    });
```

Comment:
I still prefer writing it in the alternate way because currently, the code reads like "try to set MediaQueryData, and then fail afterwards" in the try block. By putting the call to `fail` outside the try/catch blocks, it feels a lot more like "If the control flow makes it here, something's wrong" 
```suggestion
      try {
        MediaQueryData(size: null);
      } on AssertionError catch (error) {
        expect(error.toString(), contains(generateErrorMessage('size')));
        return;
      }
      fail('The assert was never called when it should have been');
```

Reply:
This makes sense.